avm/res/container-service/managed-cluster/main.bicep

```
@@ -628,6 +638,10 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2024-03-02-p
       keda: {
         enabled: kedaAddon
       }
+      verticalPodAutoscaler: {
+        addonAutoscaling: vpaAddonAutoscaling
+        enabled: vpaAddon
```

Based on my earlier comment we can set this to something like this: 
```suggestion
        enabled: (vpaAddonAutoscaling == "Enabled") ? 'enabled' : 'disabled'
```
That way this feature can be controlled by one param instead of two.

---

Looking at the API docs addonAutoscaling also enables vpa add-on but also configures vpa for the AKS addona. You might not always want that so I did it separately.

The API docs say "Whether VPA add-on is enabled and configured to scale AKS-managed add-ons."

I can do some further testing to confirm if you like.

Thoughts?

---

@PixelRobots Not sure I follow. Are you saying that we should add a check to see if addonAutoscaling is enabled before enabling vpaAddon?

---

I believe they are independent of each other. 

Both will enable vpa. One will also configure the AKS managed addons, think CSI drivers, etc to make use of the vpa.

---

So what happens if I set my params this way: 
```bicep
vpaAddon = false
vpaAddonAutoscaling = 'Enabled'
```
Should a check be added to ensure vpaAddon is `true` before configuring `addonAutoscaling`?

---

Seems addonAutoscaling is in preview, so i have removed it.